contracts/utils/Cruzo1155Factory.sol

```diff
@@ -0,0 +1,73 @@
+//SPDX-License-Identifier: MIT
+pragma solidity ^0.8.6;
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/utils/Counters.sol";
+import "@openzeppelin/contracts/proxy/beacon/BeaconProxy.sol";
+import "@openzeppelin/contracts/utils/Context.sol";
+
+contract Cruzo1155Factory is Context, Ownable {
+    using Counters for Counters.Counter;
+    Counters.Counter private _tokenIds;
+
+    event newTokenCreated(
+        address indexed tokenAddress,
+        uint256 indexed newTokenId
+    );
+
+    mapping(uint256 => address) private tokens;
+
+    address private immutable beacon;
+
+    bytes4 public selector;
+    string public baseUri;
+    address public marketAddress;
+
+    constructor(
+        address _beacon,
+        string memory rawFuncInit,
+        string memory initBaseUri,
+        address initMarketAddress
+    ) {
+        beacon = _beacon;
+        selector = bytes4(keccak256(bytes(rawFuncInit)));
+        marketAddress = initMarketAddress;
+        baseUri = initBaseUri;
+    }
+
+    function create(string calldata _name, string calldata _symbol)
```

Comment:
Please add `contractURI` argument for `create` function

Reply:
added